diff --git a/bin/arrow_ardupilot_planner.cpp b/bin/arrow_ardupilot_planner.cpp
index 6132a49..0e2dd66 100644
--- a/bin/arrow_ardupilot_planner.cpp
+++ b/bin/arrow_ardupilot_planner.cpp
@@ -385,11 +385,11 @@ int main(int argc, char* argv[]) {
   spdlog::trace("asio io_context setup");
 
   tcp::socket ap_socket(io_ctx);
-  // ap_socket.connect(*tcp::resolver(io_ctx).resolve("0.0.0.0", "5762", tcp::resolver::passive));
-  // auto mi = std::make_shared<MavlinkInterface<tcp::socket>>((std::move(ap_socket)));
-  asio::serial_port ap_serial(io_ctx, args.get("ardupilot"));
-  ap_serial.set_option(asio::serial_port_base::baud_rate(921600));
-  auto mi = std::make_shared<MavlinkInterface<asio::serial_port>>(std::move(ap_serial));
+  ap_socket.connect(*tcp::resolver(io_ctx).resolve("0.0.0.0", "5762", tcp::resolver::passive));
+  auto mi = std::make_shared<MavlinkInterface<tcp::socket>>((std::move(ap_socket)));
+  // asio::serial_port ap_serial(io_ctx, args.get("ardupilot"));
+  // ap_serial.set_option(asio::serial_port_base::baud_rate(921600));
+  // auto mi = std::make_shared<MavlinkInterface<asio::serial_port>>(std::move(ap_serial));
 
 
   auto [rs_pipe, fovh, fovv] = *setup_device().or_else([] (std::error_code e) {
