diff --git a/bin/arrow_ardupilot_planner.cpp b/bin/arrow_ardupilot_planner.cpp
index 975398e..1f8f278 100644
--- a/bin/arrow_ardupilot_planner.cpp
+++ b/bin/arrow_ardupilot_planner.cpp
@@ -286,8 +296,11 @@ int main(int argc, char* argv[]) {
   spdlog::trace("asio io_context setup");
 
   tcp::socket ap_socket(io_ctx);
-  ap_socket.connect(*tcp::resolver(io_ctx).resolve("0.0.0.0", "5762", tcp::resolver::passive));
-  auto mi = std::make_shared<MavlinkInterface<tcp::socket>>((std::move(ap_socket)));
+  // ap_socket.connect(*tcp::resolver(io_ctx).resolve("0.0.0.0", "5762", tcp::resolver::passive));
+  // auto mi = std::make_shared<MavlinkInterface<tcp::socket>>((std::move(ap_socket)));
+  asio::serial_port ap_serial(io_ctx, args.get("ardupilot"));
+  auto mi = std::make_shared<MavlinkInterface<asio::serial_port>>(std::move(ap_serial));
+
 
   auto [rs_pipe, fovh, fovv] = *setup_device().or_else([] (std::error_code e) {
     spdlog::error("Couldn't setup realsense device: {}", e.message());
